### PR TITLE
Switch FCGI from submodule to subtree

### PR DESCRIPTION
**Pull request by droark**
In Git, submodules can be a major annoyance. Move FCGI from submodule to subtree.

### CODE REVIEW

**Comment by droark** [comment]
Need to confirm using the tool as mentioned and also add a commit that removes *git submodule* from any build tools/notes. PR should be ready to go after that.

**Comment by droark** [comment]
Ran Core's [git-subtree-check.sh](https://github.com/bitcoin/bitcoin/blob/v0.16.0rc2/contrib/devtools/git-subtree-check.sh) against the FCGI subtree (`./git-subtree-check.sh cppForSwig/fcgi`). It passed, meaning this code is an exact match for what's in the parent subtree.

Going to make a couple of minor docs changes, and then this PR will be ready to go.

**Comment by droark** [comment]
PR should be ready to go now.

**Comment by droark** [comment]
In case anybody comes across this and wonders why I've submitted this PR, [this blog post](https://alistra.ghost.io/2014/11/30/git-subtree-a-better-alternative-to-git-submodule/) is far from a definitive `git submodule` vs. `git subtree` guide. However, it aptly sums up the fact that submodules are a real pain in the butt. For example, if you decide to mix submodules and subtrees, things can get ugly really quickly. Subtrees may require slightly more dev time to deal with when making changes to subtrees but, in general, they're pretty painless. That and it gets around that awful GitHub bug that prevents the submodule source code from being included when you hit the "Download ZIP" button. (Granted, users _may_ still have to rely on goatpig's specific source upload if they want to use his verification scheme, but that's fine.)

In other words, the general consensus is that submodules shouldn't be used unless you have very specific reasons for doing so. Armory doesn't (AFAIK), so it's time to [nuke them from orbit](https://www.youtube.com/watch?v=aCbfMkh940Q).

**Comment by droark** [comment]
Uggh. Borked a push just now. Need to unwind what I did.
